sources/packages/api/src/route-controllers/student-scholastic-standings/student-scholastic-standings.institutions.controller.ts

```diff
@@ -0,0 +1,113 @@
+  FormService,
+  INVALID_OPERATION_IN_THE_CURRENT_STATUS,
+  StudentAssessmentService,
+  StudentScholasticStandingsService,
+} from "../../services";
+import { ApiProcessError, ClientTypeBaseRoute } from "../../types";
+import { CustomNamedError } from "../../utilities";
+import BaseController from "../BaseController";
+import { FormNames } from "../../services/form/constants";
+import { APPLICATION_CHANGE_NOT_ELIGIBLE } from "../../constants";
+import { ScholasticStandingAPIInDTO } from "./models/student-scholastic-standings.dto";
+
+/**
+ * Scholastic standing controller for institutions Client.
+ */
+@AllowAuthorizedParty(AuthorizedParties.institution)
+@Controller("scholastic-standing")
+@ApiTags(`${ClientTypeBaseRoute.Institution}-scholastic-standing`)
+export class ScholasticStandingInstitutionsController extends BaseController {
+  constructor(
+    private readonly formService: FormService,
+    private readonly studentScholasticStandingsService: StudentScholasticStandingsService,
+    private readonly studentAssessmentService: StudentAssessmentService,
+  ) {
+    super();
+  }
+
+  /**
+   * Save scholastic standing and create new assessment.
+   * @param locationId location id to check whether the requested user and the requested application has the permission to this location.
+   * @param applicationId application id.
+   * @UserToken institution user token
+   * @param payload Scholastic Standing payload.
+   */
+  @ApiBadRequestResponse({ description: "Invalid form data." })
+  @ApiUnprocessableEntityResponse({
+    description:
+      "Application not found or invalid application or invalid" +
+      " application status or another assessment already in progress.",
+  })
+  @HasLocationAccess("locationId")
+  @Post("location/:locationId/application/:applicationId")
+  async saveScholasticStanding(
+    @Param("locationId", ParseIntPipe) locationId: number,
+    @Param("applicationId", ParseIntPipe) applicationId: number,
+    @Body() payload: ScholasticStandingAPIInDTO,
+    @UserToken() userToken: IInstitutionUserToken,
+  ): Promise<void> {
+    try {
+      const submissionResult = await this.formService.dryRunSubmission(
+        FormNames.ReportScholasticStandingChange,
+        payload.data,
+      );
+
+      if (!submissionResult.valid) {
+        throw new BadRequestException("Invalid submission.");
+      }
+      const scholasticStanding =
+        await this.studentScholasticStandingsService.processScholasticStanding(
+          locationId,
+          applicationId,
+          userToken.userId,
+          submissionResult.data.data,
+        );
+
+      // Start assessment.
+      if (scholasticStanding.studentAssessment) {
+        await this.studentAssessmentService.startAssessment(
+          scholasticStanding.studentAssessment.id,
+        );
+      }
+    } catch (error: unknown) {
+      if (error instanceof CustomNamedError) {
+        switch (error.name) {
+          case APPLICATION_NOT_FOUND:
```

Comment:
Because `:applicationId` is part of the URI, should `APPLICATION_NOT_FOUND` be a not found exception?

Reply:
Agree 👍  not part of this PR. but i will adjust it

Reply:
not part of this PR.  This error is throwing from the service, 
![image](https://user-images.githubusercontent.com/77353155/173705460-51e711e5-f3b2-4863-aa5c-5e6eb1ffa4cf.png), but the code `APPLICATION_NOT_FOUND` doesn't seem to be logical. i will update the code